python/hsfs/feature_group.py

```diff
@@ -21,12 +21,197 @@
 import numpy as np
 from typing import Optional, Union, Any, Dict, List, TypeVar
 
-from hsfs import util, engine, feature
-from hsfs.core import query, feature_group_engine, statistics_engine, feature_group_base
+from hsfs import util, engine, feature, storage_connector as sc
+from hsfs.core import (
+    feature_group_engine,
+    statistics_engine,
+    feature_group_base_engine,
+    on_demand_feature_group_engine,
+)
 from hsfs.statistics_config import StatisticsConfig
+from hsfs.constructor import query, filter
+from hsfs.client.exceptions import FeatureStoreException
 
 
-class FeatureGroup(feature_group_base.FeatureGroupBase):
+class FeatureGroupBase:
```

Comment:
I don't understand this refactor. Why putting everything on a single file?

Reply:
I am not a huge fan either, but somehow I was getting circular imports because I created the new `constructor` package. It's quite a pain to debug those, so I can try and find a different solution, but this was the most obvious fix.